### PR TITLE
Make agent log file configurable

### PR DESCRIPTION
**Pull request by ccodreanu**
### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Comment by ccodreanu** [comment]
closes #1261

**Review by nammn — COMMENTED** [review]
LGTM! Can you add a test and fix the existing ones?

**Comment by ccodreanu** [comment]
> LGTM! Can you add a test and fix the existing ones?

Thanks, done!

File: controllers/construct/mongodbstatefulset.go [hunk 1]
```diff
@@ -53,13 +54,14 @@ const (
 	VersionUpgradeHookImageEnv      = "VERSION_UPGRADE_HOOK_IMAGE"
 	ReadinessProbeImageEnv          = "READINESS_PROBE_IMAGE"
 	agentLogLevelEnv                = "AGENT_LOG_LEVEL"
+	agentLogFileEnv                 = "AGENT_LOG_FILE"
```
Comment (nammn) [inline controllers/construct/mongodbstatefulset.go:57]: What is the reason that you setup a environment variable for this? Does the agent require this?

File: controllers/construct/mongodbstatefulset.go [hunk 2]
```diff
@@ -53,13 +54,14 @@ const (
...
 	agentMaxLogFileDurationHoursEnv = "AGENT_MAX_LOG_FILE_DURATION_HOURS"
 
 	automationMongodConfFileName = "automation-mongod.conf"
 	keyfileFilePath              = "/var/lib/mongodb-mms-automation/authentication/keyfile"
 
 	automationAgentOptions    = " -skipMongoStart -noDaemonize -useLocalMongoDbTools"
-	automationAgentLogOptions = " -logFile /var/log/mongodb-mms-automation/automation-agent.log -maxLogFileDurationHrs ${AGENT_MAX_LOG_FILE_DURATION_HOURS} -logLevel ${AGENT_LOG_LEVEL}"
+	automationAgentLogOptions = " -logFile ${AGENT_LOG_FILE} -maxLogFileDurationHrs ${AGENT_MAX_LOG_FILE_DURATION_HOURS} -logLevel ${AGENT_LOG_LEVEL}"
```
Comment (nammn) [inline controllers/construct/mongodbstatefulset.go:64]: Can't you just change the method `AutomationAgentCommand()` to take the `logFile` instead and use it here?

File: controllers/construct/mongodbstatefulset.go (hunk 1 above)
Comment (nammn) [reply controllers/construct/mongodbstatefulset.go:57]: yes, you are right! I don't think this is the optimal way it has been implemented. But as you said for consistency we can keep it as it is for now

**Review by nammn — APPROVED** [review]

**Comment by jlestrange** [comment]
Hi

Any idea when this will be merged and released, we would really like to upgrade our operator with this feature?

**Pull request merged**